light-client-verifier/src/host_functions.rs

```diff
@@ -0,0 +1,81 @@
+//! Host function utilities
+
+use sp_std::fmt::Debug;
+
+/// Host functions that the light client needs for crypto operations.
+pub trait HostFunctionsProvider: Send + Sync + Default + Debug + 'static {
+    /// sha256 hash function
+    fn sha2_256(preimage: &[u8]) -> [u8; 32];
+
+    /// Verify an ed25519 signature
+    fn ed25519_verify(sig: &[u8], msg: &[u8], pub_key: &[u8]) -> bool;
+
+    /// verify secp256k1 signatures
+    fn secp256k1_verify(sig: &[u8], message: &[u8], public: &[u8]) -> bool;
```

Comment:
Using a `bool` for signature verification is a dangerous API which has lead to cryptographic vulnerabilities in practice:

https://github.com/libp2p/rust-libp2p/pull/1127/files

The [`signature`](https://docs.rs/signature) crate, already used by tendermint-rs, provides a well thought-out type safe trait-based approach to encapsulating signing and verification providers for various digital signature algorithms.

TMKMS uses this to support hardware signers for signatures, for example (e.g. Ledger, YubiHSM2)